deep-learning/src/main/python/synapse/ml/dl/DeepTextClassifier.py

```diff
@@ -0,0 +1,306 @@
+        )
+
+        kwargs = self._kwargs
+        self._set(**kwargs)
+
+        self._update_cols()
+        self._update_transformation_fn()
+
+        model = LitDeepTextModel(
+            checkpoint=self.getCheckpoint(),
+            additional_layers_to_train=self.getAdditionalLayersToTrain(),
+            num_labels=self.getNumClasses(),
+            optimizer_name=self.getOptimizerName(),
+            loss_name=self.getLossName(),
+            label_col=self.getLabelCol(),
+            text_col=self.getTextCol(),
+            train_from_scratch=self.getTrainFromScratch(),
+        )
+        self._set(model=model)
+
+    def setCheckpoint(self, value):
+        return self._set(checkpoint=value)
+
+    def getCheckpoint(self):
+        return self.getOrDefault(self.checkpoint)
+
+    def setAdditionalLayersToTrain(self, value):
+        return self._set(additional_layers_to_train=value)
+
+    def getAdditionalLayersToTrain(self):
+        return self.getOrDefault(self.additional_layers_to_train)
+
+    def setNumClasses(self, value):
+        return self._set(num_classes=value)
+
+    def getNumClasses(self):
+        return self.getOrDefault(self.num_classes)
+
+    def setLossName(self, value):
+        return self._set(loss_name=value)
+
+    def getLossName(self):
+        return self.getOrDefault(self.loss_name)
+
+    def setOptimizerName(self, value):
+        return self._set(optimizer_name=value)
+
+    def getOptimizerName(self):
+        return self.getOrDefault(self.optimizer_name)
+
+    def setTokenizer(self, value):
+        return self._set(tokenizer=value)
+
+    def getTokenizer(self):
+        return self.getOrDefault(self.tokenizer)
+
+    def setMaxTokenLen(self, value):
+        return self._set(max_token_len=value)
+
+    def getMaxTokenLen(self):
+        return self.getOrDefault(self.max_token_len)
+
+    def setTrainFromScratch(self, value):
+        return self._set(train_from_scratch=value)
+
+    def getTrainFromScratch(self):
+        return self.getOrDefault(self.train_from_scratch)
+
+    def _update_cols(self):
+        self.setFeatureCols([self.getTextCol()])
+        self.setLabelCols([self.getLabelCol()])
+
+    def _fit(self, dataset):
+        return super()._fit(dataset)
+
+    # override this method to provide a correct default backend
+    def _get_or_create_backend(self):
+        backend = self.getBackend()
+        num_proc = self.getNumProc()
+        if backend is None:
+            if num_proc is None:
+                num_proc = self._find_num_proc()
+            backend = SparkBackend(
+                num_proc,
+                stdout=sys.stdout,
+                stderr=sys.stderr,
+                prefix_output_with_timestamp=True,
+                verbose=self.getVerbose(),
+            )
+        elif num_proc is not None:
+            raise ValueError(
+                'At most one of parameters "backend" and "num_proc" may be specified'
+            )
+        return backend
```

Comment:
Now that we have 2 transformers in this project, any way we can abstract out shared code into a base class?

Reply:
Sure.